packages/core/src/App/Containers/AccountSignupModal/account-signup-modal.jsx

```diff
@@ -180,10 +180,19 @@ const AccountSignupModal = ({
     disableApp,
     is_loading,
     is_visible,
+    is_logged_in,
+    logout,
     onSignup,
     residence_list,
     toggleAccountSignupModal,
 }) => {
+    React.useEffect(() => {
+        // a logged in user should not be able to create a new account
+        if (is_visible && is_logged_in) {
+            logout();
+        }
+    }, [is_visible, is_logged_in, logout]);
```

Comment:
Do we need `logout` in dependency array?

Reply:
@balakrishna-binary true it's not super necessary to add `logout` here, the reason for it is:
* readability -  explicitly declare what values from the component scope that are used in the effect
* avoid adding eslint ignore for breaking the hook rules

https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Reply:
@oskar-binary Thanks for the explanation 🙇